### PR TITLE
js_notification_DOM: Cypress tests added

### PR DESCRIPTION
**Pull request by iBrianWarner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,11 @@ Your task: Create function `pushNotification`, which takes coords of future mess
...
 Implementation tips:
-- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`). 
-- The message should have a title with class `title` (prefer `h2` element)
-- The message should have description (prefer tag `p`)
-- Set coordinates of the message (top and right) rely on values from parameters.
+- Print three messages (`success`, `error`, `warning`);
+- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`); 
+- The message should have a title with class `title` (prefer `h2` element);
+- The message should have description (prefer tag `p`);
+- Set coordinates of the message (top and right) rely on values from parameters. You can find coordinates in tests.
```
Comment (YuriiHoliuk) [inline README.md:18]: Не треба, щоб вони шукали в тестах все має бути зрозумілим з умови, чи бути у вхідних параметрах, якщо мова йде про функцію

File: cypress/integration/js_notification.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+///<reference types=‘cypress’/>
+'use strict';
+
+describe('Tests for your notifications.', () => {
```
Comment (YuriiHoliuk) [inline cypress/integration/js_notification.spec.js:4]: Це як моя автобіографія

File: cypress/integration/js_notification.spec.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  before(() => {
+    cy.visit('/');
+  });
+
+  it('Success message.', () => {
+    cy.get('[class = "notification success"]').should('have.attr', 'style', 'top: 10px; right: 10px;');
```
Comment (YuriiHoliuk) [inline cypress/integration/js_notification.spec.js:10]:
```suggestion
    cy.get('.notification.success').should('have.attr', 'style', 'top: 10px; right: 10px;');
```

use aliases

File: cypress/integration/js_notification.spec.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  it('Warning message.', () => {
+    cy.get('[class = "notification warning"]').should('have.attr', 'style', 'top: 290px; right: 10px;');
+    cy.get('[class = "notification warning"]').children('h2').should('have.attr', 'class', 'title');
+    cy.get('[class = "notification warning"]').children('p');
+  });
+
+  it('Messages should disappear in 2 seconds.', () => {
+    cy.wait(2000).then(() => {
```
Comment (YuriiHoliuk) [inline cypress/integration/js_notification.spec.js:28]: так як ви використовуєте before, а не beforeEach, пройшло вже значно більше 2 секунд

Comment (YuriiHoliuk) [reply cypress/integration/js_notification.spec.js:28]: prefer beforeEach hook

Comment (YuriiHoliuk) [reply cypress/integration/js_notification.spec.js:28]: але можна дивитись від випадку

File: README.md (hunk 1 above)
Comment (iBrianWarner) [reply README.md:18]: відредагував умову, щоб було зрозуміліше

File: cypress/integration/js_notification.spec.js (hunk 2 above)
Comment (iBrianWarner) [reply cypress/integration/js_notification.spec.js:4]: переписав, щоб ніхто не знав твою біографію

File: cypress/integration/js_notification.spec.js (hunk 3 above)
Comment (iBrianWarner) [reply cypress/integration/js_notification.spec.js:10]: додав aliases

File: cypress/integration/js_notification.spec.js (hunk 4 above)
Comment (iBrianWarner) [reply cypress/integration/js_notification.spec.js:28]: зробив beforeEach

File: README.md [hunk 5]
```diff
@@ -1,22 +1,30 @@
...
+    - Run `npm run test` command to test your code;
+    - Run `npm run test:only -- -n` to run fast test ignoring linter;
+    - Run `npm run test:only -- -l` to run fast test with additional info in console ignoring linter.
 
 ### Task: Create a notification
 
 To inform the user about the success of the operation, or any errors, websites usually use messages that appear for a while on the screen.
 
-Your task: Create function `pushNotification`, which takes coords of future message, title, description and type (warning, success or error), appends it to the document and remove it after 2 seconds.
+Your task: Create function `pushNotification`, which takes coords of future message, title, description and type
```
Comment (roman-mate) [inline README.md:12]:
```suggestion
Your task: Create function `pushNotification`, which takes coordinates of the message, title, description, and type 
```

File: README.md [hunk 6]
```diff
@@ -1,22 +1,30 @@
...
+    - Run `npm run test:only -- -n` to run fast test ignoring linter;
+    - Run `npm run test:only -- -l` to run fast test with additional info in console ignoring linter.
 
 ### Task: Create a notification
 
 To inform the user about the success of the operation, or any errors, websites usually use messages that appear for a while on the screen.
 
-Your task: Create function `pushNotification`, which takes coords of future message, title, description and type (warning, success or error), appends it to the document and remove it after 2 seconds.
+Your task: Create function `pushNotification`, which takes coords of future message, title, description and type 
+(success, error and warning), appends it to the document and remove it after 2 seconds.
```
Comment (roman-mate) [inline README.md:13]:
```suggestion
(success, error and warning). `pushNotification` creates an element to display the message, appends it to the document, and removes it from the DOM after 2 seconds.
```

File: README.md [hunk 7]
```diff
@@ -1,22 +1,30 @@
...
 You no need to write any styles or HTML code in this task. Change only `main.js` file.
```
Comment (roman-mate) [inline README.md:14]:
```suggestion
Do not write any CSS styles or HTML code in this task. You should modify only `main.js` file.
```

File: README.md [hunk 8]
```diff
@@ -1,22 +1,30 @@
...
+messages: 
+        - `success`: 10, 10; 
+        - `error`: 150, 10; 
+        - `warning`: 290, 10. 
 
 Hint: use [setTimeout](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) for execute a function or specified piece of code once the timer expires.
 
 It's an expected result of your job:
-![Result](./src/images/result.png)
+![Result](./src/images/result_new.png)
```
Comment (roman-mate) [inline README.md:30]: why rename the file? Just replace the content.

File: README.md [hunk 9]
```diff
@@ -1,22 +1,30 @@
...
+Your task: Create function `pushNotification`, which takes coords of future message, title, description and type 
+(success, error and warning), appends it to the document and remove it after 2 seconds.
 You no need to write any styles or HTML code in this task. Change only `main.js` file.
 
 Implementation tips:
-- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`). 
-- The message should have a title with class `title` (prefer `h2` element)
-- The message should have description (prefer tag `p`)
-- Set coordinates of the message (top and right) rely on values from parameters.
+- Print three messages (`success`, `error`, `warning`);
```
Comment (roman-mate) [inline README.md:17]:
```suggestion
- Print three messages: `success`, `error`, `warning` (see image below);
```

File: README.md [hunk 10]
```diff
@@ -1,22 +1,30 @@
...
+- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`); 
+- The message should have a title with class `title` (prefer `h2` element);
+- The message should have description (prefer tag `p`);
+- Set coordinates of the message (`top` and `right`) rely on values from parameters. Use this coordinates for your
+messages: 
+        - `success`: 10, 10; 
+        - `error`: 150, 10; 
+        - `warning`: 290, 10.
```
Comment (roman-mate) [inline README.md:25]: who calls the pushNotification method?  Should we add the call in solution template?

**Review by roman-mate — CHANGES_REQUESTED** [review]

File: README.md (hunk 10 above)
Comment (iBrianWarner) [reply README.md:25]: calls were added to the solution template

**Review by iBrianWarner — COMMENTED** [review]
the picture was renamed
added helper to the tests
added calls to the solution

File: src/scripts/main.js [hunk 11]
```diff
@@ -3,3 +3,15 @@
 // const pushNotification = (top, right, title, description, type) => {
 //   // write code here
 // };
+
+// pushNotification(10, 10, 'Title of Success message',
+//   'This example of result.\n '
```
Comment (roman-mate) [inline src/scripts/main.js:8]:
```suggestion
//   'Message example.\n '
```

File: README.md [hunk 12]
```diff
@@ -1,20 +1,28 @@
...
+Your task: Create function `pushNotification`, which takes coordinates of the message, title, description, and type 
+(success, error and warning). `pushNotification` creates an element to display the message, appends it to the document, and removes it from the DOM after 2 seconds.
+Do not write any CSS styles or HTML code in this task. You should modify only `main.js` file.
 
 Implementation tips:
-- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`). 
-- The message should have a title with class `title` (prefer `h2` element)
-- The message should have description (prefer tag `p`)
-- Set coordinates of the message (top and right) rely on values from parameters.
+- Print three messages: `success`, `error`, `warning` (see image below);
```
Comment (roman-mate) [inline README.md:17]: explain that function calls are already in main.js

File: README.md [hunk 13]
```diff
@@ -1,20 +1,28 @@
...
+- The message is block element with class `notification` + class which depends on the input parameter type (`success`, `error`, `warning`); 
+- The message should have a title with class `title` (prefer `h2` element);
+- The message should have description (prefer tag `p`);
+- Set coordinates of the message (`top` and `right`) rely on values from parameters. Use this coordinates for your
+messages: 
+        - `success`: 10, 10; 
+        - `error`: 150, 10; 
+        - `warning`: 290, 10.
```
Comment (roman-mate) [inline README.md:25]: is this section needed?

File: src/scripts/main.js [hunk 14]
```diff
@@ -3,3 +3,15 @@
...
+//   + 'Notification should contain title and description.', 'success');
+
+// pushNotification(150, 10, 'Title of Error message',
+//   'This example of result.\n '
+//   + 'Notification should contain title and description.', 'error');
+
+// pushNotification(290, 10, 'Title of Warning message',
+//   'This example of result.\n '
+//   + 'Notification should contain title and description.', 'warning');
```
Comment (roman-mate) [inline src/scripts/main.js:17]: let's uncomment these lines?

File: src/scripts/main.js (hunk 11 above)
Comment (iBrianWarner) [reply src/scripts/main.js]: done

File: README.md (hunk 12 above)
Comment (iBrianWarner) [reply README.md]: done

File: README.md (hunk 13 above)
Comment (iBrianWarner) [reply README.md]: agree. deleted

File: src/scripts/main.js (hunk 14 above)
Comment (iBrianWarner) [reply src/scripts/main.js]: uncommented

**Review by brainwashers — APPROVED** [review]

File: src/scripts/main.js [hunk 15]
```diff
@@ -1,5 +1,17 @@
 'use strict';
 
-// const pushNotification = (top, right, title, description, type) => {
-//   // write code here
-// };
+const pushNotification = (topPst, rightPst, title, description, type) => {
```
Comment (roman-mate) [inline src/scripts/main.js:3]: I have not idea what 'Pst' stands for. Perhaps 'position'? Pos is a standard abbreviation for position. 

top, right should be sufficient. Why have you decided to rename them?

Comment (iBrianWarner) [reply src/scripts/main.js:3]: image
I've decided to rename the abbreviation as the linter said that such value already declared and I can't find where it was.

Comment (roman-mate) [reply src/scripts/main.js]: chatted offline. top is global from window.top. keeping unique name.

**Review by roman-mate — APPROVED** [review]

**Pull request merged**